### PR TITLE
Feature/test favorites watchlist

### PR DESCRIPTION
**Pull request by cyberspacedk**

### CODE REVIEW

File: src/components/Favorites/__test__/component.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    history: {
+      push: () => {},
+    },
+    operationsFavoritesRequest: () => {},
+    goToNextPage: jest.fn(),
+  };
+
+  it('Match its snapshot. Array provided', () => {
+    const wrapper = shallow(
+      <Favorites {...props} title="Favorites" removeBox />,
```
Comment (plzen) [inline src/components/Favorites/__test__/component.test.js]: Move `title` and `removeBox` props to `props` object

File: src/components/Favorites/__test__/container.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    instance.componentDidMount();
+    expect(store.dispatch).toHaveBeenCalledWith(getFavoritesRequest());
+  });
+
+  it('Check is calling container method.', () => {
+    const wrapper = mount(<FavoritesContainer store={store} />);
+    const container = wrapper.find('FavoritesContainer').instance();
+    jest.spyOn(container, 'goToNextPage');
+    container.goToNextPage();
+    expect(container.goToNextPage).toHaveBeenCalled();
```
Comment (plzen) [inline src/components/Favorites/__test__/container.test.js]: This test checking nothing, you need to call `goToNextPage` and verify that it calls `getFavoritesRequest` with params

File: src/components/Home/Dashboard/Content/TrendingMovies/__test__/component.test.js [hunk 3]
```diff
@@ -1,28 +1,48 @@
...
+    error: false,
+    empty: false,
   };
 
-  it('Match its snapshot. Display Data from DataBse', () => {
-    const TopFilmsComponent = shallow(<TopFilms {...props} />);
-    expect(shallowToJson(TopFilmsComponent)).toMatchSnapshot();
+  it('Match its snapshot. Array provided', () => {
+    const wrapper = shallow(<TrendingMovies {...props} />);
+    expect(shallowToJson(wrapper)).toMatchSnapshot();
```
Comment (plzen) [inline src/components/Home/Dashboard/Content/TrendingMovies/__test__/component.test.js]: Don't need to use `shallowToJson` - it's already added to config.

File: src/components/Home/Dashboard/Content/TrendingMovies/__test__/component.test.js [hunk 4]
```diff
@@ -1,28 +1,48 @@
...
+    const nextProps = {
+      ...props,
+      topFilms: [],
+      error: true,
+    };
+    const wrapper = shallow(<TrendingMovies {...nextProps} />);
+    expect(shallowToJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('Match its snapshot. Empty', () => {
```
Comment (plzen) [inline src/components/Home/Dashboard/Content/TrendingMovies/__test__/component.test.js]: This test is the same as previous, do you need it?

File: src/components/Home/Dashboard/Content/TrendingMovies/__test__/container.test.js [hunk 5]
```diff
@@ -1,38 +1,32 @@
...
   it('Should match its snapshot', () => {
-    expect(shallowToJson(container)).toMatchSnapshot();
+    expect(container).toMatchSnapshot();
   });
 
-  // ПРИМЕР SPY
-  it('Check call lifeCycleMethod componentDidMount', () => {
-    mount(<TopFilmsContainerConnected store={store} />);
-    // const spy = jest.spyOn(store, 'dispatch');
+  xit('Check call lifeCycleMethod componentDidMount', () => {
```
Comment (plzen) [inline src/components/Home/Dashboard/Content/TrendingMovies/__test__/container.test.js:28]: Enable test

File: src/components/Watchlist/__test__/container.test.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  const wrapper = shallow(<WatchListContainer store={store} />);
+  const container = wrapper.dive().dive();
+  const instance = container.instance();
+
+  it('Should match its snapshot', () => {
+    expect(container).toMatchSnapshot();
+  });
+
+  it('Check call lifeCycleMethod componentDidMount', () => {
+    mount(<WatchListContainer store={store} />);
```
Comment (plzen) [inline src/components/Watchlist/__test__/container.test.js]: Don't use mount in tests, your next test doing the same without mount

File: src/components/Watchlist/__test__/container.test.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  it('should call getWatchListRequest action', () => {
+    instance.componentDidMount();
+    expect(store.dispatch).toHaveBeenCalledWith(getWatchListRequest());
+  });
+
+  it('Check is calling container method.', () => {
+    const wrapper = mount(<WatchListContainer store={store} />);
+    const container = wrapper.find('WatchListContainer').instance();
+    jest.spyOn(container, 'goToNextPage');
+    container.goToNextPage();
```
Comment (plzen) [inline src/components/Watchlist/__test__/container.test.js]: This test doing nothing - you need to check if calling `goToNextPage` calls another method

File: src/components/shared/Layout/__test__/component.test.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  it('Match its snapshot. Error', () => {
+    const nextProps = {
+      ...props,
+      error: true,
+    };
+    const wrapper = shallow(<PageLayout {...nextProps} />);
+    expect(shallowToJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('Match its snapshot. Empty', () => {
```
Comment (plzen) [inline src/components/shared/Layout/__test__/component.test.js]: The same test as previous

File: src/store/authentifiction/__test__/action._test.js [hunk 9]
```diff
@@ -1,4 +1,10 @@
-import { userLogout, authUser, setUserData } from '../actions';
+/* import {
```
Comment (plzen) [inline src/store/authentifiction/__test__/action._test.js]: Remove commented code

File: src/store/favorites/__test__/logic.test.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    dispatch,
+    done,
+    getState,
+  );
+
+  it('Check: is all actions were dispatched', () => {
+    expect(dispatch.mock.calls.length).toBe(2);
+  });
+
+  xit('dispatches action - WRITE_TO_DATABASE and GET_FAVORITES_RESPONSE', () => {
```
Comment (plzen) [inline src/store/favorites/__test__/logic.test.js]: Enable test

File: src/store/favorites/__test__/logic.test.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+      },
+    });
+  });
+
+  it('calls done', () => {
+    expect(done).toBeCalled();
+  });
+});
+
+describe('Favorites: operationsFavoriteLogic', () => {
```
Comment (plzen) [inline src/store/favorites/__test__/logic.test.js:68]: Why don't you test network request?

File: src/store/favorites/__test__/reducers.test.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+describe('Favorites: reducers', () => {
+  it('should set loading field to true', () => {
+    const action = {
+      type: 'GET_FAVORITES_REQUEST',
+    };
+
+    const state = {
+      ...initialState,
+      loading: false,
+      error: false,
```
Comment (plzen) [inline src/store/favorites/__test__/reducers.test.js]: You don't need error here if your reducer doesn't change this value

File: src/store/favorites/__test__/selectors.test.js [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    });
+
+    it('returns false if data exist', () => {
+      expect(isEmpty(state)).toBeFalsy();
+    });
+  });
+
+  describe('Favorites: current page', () => {
+    it('returns current page ', () => {
+      expect(getCurrentPage(state)).toEqual(state.favorites.current_page);
```
Comment (plzen) [inline src/store/favorites/__test__/selectors.test.js]: Don't use state (like `state.favorites.current_page`) in matchers, use values

File: src/store/favorites/__test__/selectors.test.js [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+    });
+  });
+
+  describe('Favorites: total pages', () => {
+    it('returns all pages ', () => {
+      expect(getTotalPages(state)).toEqual(state.favorites.total_results);
```
Comment (plzen) [inline src/store/favorites/__test__/selectors.test.js]: Don't use state (like `state.favorites.current_page`) in matchers, use values

File: src/store/trendingMovies/__test__/logic.test.js [hunk 15]
```diff
@@ -21,10 +21,10 @@ describe('getTopFilmsLogic operation', () => {
 
   it('dispatches action - FETCH_RESPONSE', () => {
     expect(dispatch.mock.calls.length).toBe(1);
-    expect(dispatch.mock.calls[0][0]).toEqual({
+    /*  expect(dispatch.mock.calls[0][0]).toEqual({
```
Comment (plzen) [inline src/store/trendingMovies/__test__/logic.test.js]: Remove commented code

File: src/store/trendingMovies/__test__/logic.test.js [hunk 16]
```diff
@@ -1,18 +1,18 @@
...
+    response: { data: [1, 2, 3] },
   });
 
   const getState = jest.fn();
   const done = jest.fn();
   const dispatch = jest.fn();
 
   getState.mockReturnValue({
-    films: [{}],
+    ids: [1, 2, 3],
```
Comment (plzen) [inline src/store/trendingMovies/__test__/logic.test.js]: Why do you need this values in state??

File: src/store/trendingMovies/__test__/reducer.test.js [hunk 17]
```diff
@@ -16,20 +20,32 @@ describe('reducers', () => {
...
       type: 'FETCH_RESPONSE',
-      payload: [{}],
+      payload: [1, 2, 3],
     };
-    expect(reducer(initialState, action)).toEqual({
+    const state = {
       ...initialState,
-      films: action.payload,
+      loading: true,
+      error: false,
```
Comment (plzen) [inline src/store/trendingMovies/__test__/reducer.test.js]: Don't use fields which reducer doesn't change

File: src/store/watchList/__test__/reducers.test.js [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+
+describe('Watchlist: reducers', () => {
+  it('should set loading field to true', () => {
+    const action = {
+      type: 'GET_WATCHLIST_REQUEST',
+    };
+    const state = {
+      ...initialState,
+      loading: false,
+      error: false,
```
Comment (plzen) [inline src/store/watchList/__test__/reducers.test.js]: Don't use field which reducer doesn't change

**Review by plzen — CHANGES_REQUESTED** [review]

File: src/components/Home/Dashboard/Content/TrendingMovies/__test__/container.test.js (hunk 5 above)
Comment (cyberspacedk) [reply src/components/Home/Dashboard/Content/TrendingMovies/__test__/container.test.js:28]: I m going to testing lifecycle and method after all.

File: src/components/Favorites/__test__/container.test.js [hunk 19]
```diff
@@ -36,9 +36,12 @@ describe('FavoritesContainer ', () => {
   it('Check is calling container method.', () => {
     const wrapper = mount(<FavoritesContainer store={store} />);
     const container = wrapper.find('FavoritesContainer').instance();
-    jest.spyOn(container, 'goToNextPage');
+    jest.spyOn(store, 'dispatch');
```
Comment (plzen) [inline src/components/Favorites/__test__/container.test.js]: Dispatch is already mocked https://github.com/cyberspacedk/MovieDB/pull/11/files/c08d388142d6dfa45e7bc526be8bd773d8b83b80..a73258b91a5f147f40f20ab7d6dac37a88880ad9#diff-06ce269d62790f4c785a654d343f7bf2R21

File: src/components/Watchlist/__test__/container.test.js [hunk 20]
```diff
@@ -29,7 +29,6 @@ describe('WatchListContainer ', () => {
   });
 
   it('Check call lifeCycleMethod componentDidMount', () => {
-    mount(<WatchListContainer store={store} />);
     expect(store.dispatch).toHaveBeenCalledWith(getWatchListRequest());
```
Comment (plzen) [inline src/components/Watchlist/__test__/container.test.js]: This code is tested in the next `it` block

File: src/components/Watchlist/__test__/container.test.js [hunk 21]
```diff
@@ -38,14 +37,6 @@ describe('WatchListContainer ', () => {
...
   });
 
-  it('Check is calling container method.', () => {
```
Comment (plzen) [inline src/components/Watchlist/__test__/container.test.js]: Why do you remove this test? You need to cover `goToNextPage` method

File: src/store/favorites/__test__/logic.test.js [hunk 22]
```diff
@@ -34,21 +34,9 @@ describe('Favorites: getFavoritesLogic', () => {
...
-      },
-    });
-    expect(dispatch.mock.calls[0][1]).toEqual({
-      type: 'GET_FAVORITES_RESPONSE',
-      payload: {
-        ids: [1],
-      },
-    });
+  it('dispatches action - WRITE_TO_DATABASE and GET_FAVORITES_RESPONSE', () => {
+    expect(dispatch.mock.calls[0][0].type).toEqual('WRITE_TO_DATABASE');
```
Comment (plzen) [inline src/store/favorites/__test__/logic.test.js]: Why don't you test payload?

**Review by plzen — CHANGES_REQUESTED** [review]

**Review by plzen — APPROVED** [review]

**Pull request merged**